noir_stdlib/src/array/mod.nr

```diff
@@ -5,6 +5,24 @@ use crate::runtime::is_unconstrained;
 mod check_shuffle;
 mod quicksort;
 
+/// Creates a new array containing numbers from 0 to N-1.
+/// This is a utility function to create arrays from ranges.
+///
+/// Example:
+/// ```noir
+/// fn main() {
+///     let arr = std::array::range(5);
+///     assert(arr == [0, 1, 2, 3, 4]);
+/// }
+/// ```
+pub fn range<T>(size: Field) -> [Field] {
```

Comment:
The `T` generic parameter is not used, and technically this returns a [slice](https://github.com/noir-lang/noir/blob/1cb40881b0de2d9fd5ac1700cbb78cead6e49097/noir_stdlib/src/slice.nr), rather than an array. 
`size` is usually expressed as `u32`.
Returning `Field` seems a bit specific; the library so far is completely generic.

Perhaps something like [std::array::from_fn](https://doc.rust-lang.org/nightly/core/array/fn.from_fn.html) would be a more flexible alternative to make an actual array of a given size, with content created for each index given to a lambda as `u32` between 0 and `N`. You could use it to cast to `Field`, or map it to a range other than 0..N, e.g. by an offset.